src/Common/src/System/Collections/Generic/ComparerUtilities.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Generic
+{
+    internal static class ComparerUtilities
+    {
+        public static bool AreComparersEqual<T>(IComparer<T> left, IComparer<T> right)
+        {
+            if (left == right)
+            {
+                return true;
+            }
+
+            var @default = Comparer<T>.Default;
```

Comment:
nit: rename this to defaultComparer. It removes the need to have @ in the variable name.